lib/Doctrine/DBAL/Schema/Table.php

```diff
@@ -689,22 +723,6 @@ public function getPrimaryKey()
     }
 
     /**
-     * Returns the primary key columns.
-     *
-     * @return array
-     *
-     * @throws DBALException
-     */
-    public function getPrimaryKeyColumns()
-    {
-        if ( ! $this->hasPrimaryKey()) {
-            throw new DBALException("Table " . $this->getName() . " has no primary key.");
```

Comment:
This exception is gone - that is a BC break, as now an empty list of primary key columns is returned